File: Server/src/main/content/global/skill/production/crafting/plugins/GemCutPlugin.java
package content.global.skill.production.crafting.plugins;

import content.global.skill.production.crafting.data.GemData;
import content.global.skill.production.crafting.item.GemCutPulse;
import core.game.dialogue.SkillDialogueHandler;
import core.game.dialogue.SkillDialogueHandler.SkillDialogue;
import core.game.interaction.NodeUsageEvent;
import core.game.interaction.UseWithHandler;
import core.game.node.entity.player.Player;
import core.plugin.Initializable;
import core.plugin.Plugin;

@Initializable
public final class GemCutPlugin extends UseWithHandler {

    public GemCutPlugin() {
        super(1623, 1621, 1619, 1617, 1631, 6571, 1625, 1627, 1629, 6571);
    }

    @Override
    public Plugin<Object> newInstance(Object arg) throws Throwable {
        addHandler(1755, ITEM_TYPE, this);
        return null;
    }

    @Override
    public boolean handle(NodeUsageEvent event) {
        final Player player = event.getPlayer();
        final GemData gem = GemData.forId(event.getUsedItem().getId() == 1755 ? event.getBaseItem() : event.getUsedItem());
        SkillDialogueHandler handler = new SkillDialogueHandler(player, SkillDialogue.ONE_OPTION, gem.getGem()) {
            @Override
            public void create(final int amount, int index) {
                player.getPulseManager().run(new GemCutPulse(player, gem.getUncut(), amount, gem));
            }

            @Override
            public int getAll(int index) {
                return player.getInventory().getAmount(gem.getUncut());
            }

        };
        if (player.getInventory().getAmount(gem.getUncut()) == 1) {
            handler.create(0, 1);
        } else {
            handler.open();
        }
        return true;
    }

}

File: Server/src/main/content/global/skill/production/crafting/plugins/GemCutPlugin.kt
package content.global.skill.production.crafting.plugins

import content.global.skill.production.crafting.data.GemData
import content.global.skill.production.crafting.item.GemCutPulse
import core.api.consts.Items
import core.game.dialogue.SkillDialogueHandler
import core.game.interaction.NodeUsageEvent
import core.game.interaction.UseWithHandler
import core.plugin.Initializable
import core.plugin.Plugin

@Initializable
class GemCutPlugin : UseWithHandler(
    Items.UNCUT_DIAMOND_1617,
    Items.UNCUT_RUBY_1619,
    Items.UNCUT_EMERALD_1621,
    Items.UNCUT_SAPPHIRE_1623,
    Items.UNCUT_OPAL_1625,
    Items.UNCUT_JADE_1627,
    Items.UNCUT_RED_TOPAZ_1629,
    Items.UNCUT_DRAGONSTONE_1631,
    Items.UNCUT_ONYX_6571
) {

    override fun newInstance(arg: Any?): Plugin<Any> {
        addHandler(1755, ITEM_TYPE, this)
        return this
    }

    override fun handle(event: NodeUsageEvent): Boolean {
        val player = event.player
        val gem = GemData.forId(if (event.usedItem.id == Items.CHISEL_1755) event.baseItem else event.usedItem)
        val handler: SkillDialogueHandler = object : SkillDialogueHandler(player, SkillDialogue.ONE_OPTION, gem.gem) {
            override fun create(amount: Int, index: Int) {
                player.pulseManager.run(GemCutPulse(player, gem.uncut, amount, gem))
            }

            override fun getAll(index: Int): Int {
                return player.inventory.getAmount(gem.uncut)
            }
        }
        if (player.inventory.getAmount(gem.uncut) == 1) {
            handler.create(0, 1)
        } else {
            handler.open()
        }
        return true
    }
}
